llvm/lib/CodeGen/CheriGetAddressElim.cpp

```diff
@@ -110,6 +110,11 @@ static bool couldReadCapMetadata(MachineInstr *MI,
 }
 
 bool CheriGetAddressElim::runOnMachineFunction(MachineFunction &MF) {
+  MVT CapTy = MF.getSubtarget().getTargetLowering()->cheriCapabilityType();
+  // Skip this pass if the function is compiler without CHERI support.
```

Comment:
Ack. Thinking about this more I think the precedent is to do it like this (and CheriBoundAllocas skips based on AllocaAS). Probably we should be good citizens and downstream the change with a corresponding AArch64 TM change to Morello LLVM (happy to do that, not difficult), otherwise it'll probably get missed in a future merge and the redundant TM code will remain.